tests/task/test_consolidate_query_matches.py

```diff
@@ -0,0 +1,50 @@
+from fidesops.graph.config import FieldPath
+from fidesops.task.consolidate_query_matches import consolidate_query_matches
+
+
+def test_consolidate_query_matches():
+    # Matching scalar returned
+    input_data = {"B": 55}
+    target_path = FieldPath("B")
+    assert consolidate_query_matches(input_data, target_path) == [55]
+
+    # Matching array returned as-is
+    input_data = {"A": [1, 2, 3]}
+    target_path = FieldPath("A")
+    assert consolidate_query_matches(input_data, target_path) == [1, 2, 3]
+
+    # Array of embedded objects have multiple matching sub-paths merged
+    field_path = FieldPath("A", "B")
+    input_data = {"A": [{"B": 1, "C": 2}, {"B": 3, "C": 4}, {"B": 5, "C": 6}]}
+    assert consolidate_query_matches(input_data, field_path) == [1, 3, 5]
+
+    # Nested array returned
+    input_data = {"A": {"B": {"C": [9, 8, 7]}}, "D": {"E": {"F"}}}
```

Comment:
Is the intention to pass `{"F"}` in as a set here?

Reply:
nope, but I was just adding extra keys to this dictionary - that path isn't crawled. I'll fix.